core/requirements/core/REQ_job-results-param-outputs.adoc

```diff
@@ -0,0 +1,23 @@
+[[req_core_job-results-param-outputs]]
+[requirement]
+====
+[%metadata]
+label:: /req/core/job-results-param-outputs
+[.component,class=part]
+--
+The operation at the `/jobs/{jobID}/results` endpoint SHALL support a parameter `outputs` with the following characteristics (using an OpenAPI Specification 3.0 fragment):
+
```

Comment:
@gfenoy there are two endpoints: `/jobs/{jobID}/results` and `/jobs/{jobID}/results/{outputID}`.  At the `/jobs/{jobID}/results/{outputID}` you only get a single output identified by the `outputID`.  At the `/jobs/{jobID}/results` endpoint you can get ALL the outputs.  But, what if you want SOME of the outputs?  To allow this, I added the `output` query parameter.  So lets say you execute a process and ask it to generate 10 outputs BUT you only want to fetch 3 of the output initially.  To handle this use case I added the `outputs` query parameter so that you can say `/jobs/{jobsID}/results?outputs=output2,output7,output9`.  I suppose I could have made the `outputID` path parameter an array so that you could say `/jobs/{jobID}/results/output2,output7.output9` but I thought this is a less common form than using a query parameter.   Perhaps I should rename the `outputs` query parameter to `outputIDs` or something like that ...

Reply:
Sorry for my misunderstanding, I thought we were defining `{outputID}` here. 

I am perfectly fine with having this new `outputs` query parameter as an  array. I hope it does not add any redundancy with the `{outputID}` path parameter.

Reply:
@gfenoy `outputs` is a query parameter and `{outputID}` is a path parameter so there should be no redundancy.  However, as I mentioned above, I could get rid of the `outputs` query parameter and make the `{outputID}` path parameter an array which would mean:

- get ALL results: /jobs/{jobID}/results
- get ONE result: /jobs/{jobID}/results/{outputID}
- get SOME results: /jobs/{jobID}/results/{outputID},{outputID},{outputID},...

My only concern with this approach is that I believe that the last form is not very common.  I think most people would expect the last one to be `/jobs/{jobID}/results?outputs=outputID,outputID,outputID` ... but I could be wrong.  I am agnostic and would be happy either way.